### PR TITLE
SQL: Fix JDBC url pattern in docs and error message

### LINKED ISSUE #56476: SQL: the invalid url error message should be in line with the documentation
We state in docs that the [url format should be](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/sql-jdbc.html#jdbc-setup) `jdbc:es://[[http|https]://]*[host[:port]]*/[prefix]*<[?[option=value]&]*` but if the url is indeed incorrect, the error message mentions `Invalid URL [jdbc:es://localhost:9200/?ssl=#5#], format should be [jdbc:es://[http|https]?[host[:port]]*/[prefix]*[?[option=value]&]*]` which is inaccurate.

### PR DESCRIPTION
**Pull request by matriv**
The docs pattern url was using `*` which means zero or many instead
of `?` which means zero or one. The pattern url returned in error
messages was not in sync with the one in the docs.

Fixes: #56476

### CODE REVIEW

File: docs/reference/sql/endpoints/jdbc.asciidoc [hunk 1]
```diff
@@ -51,7 +51,7 @@ Once registered, the driver understands the following syntax as an URL:
 
 ["source","text",subs="attributes"]
 ----
-jdbc:es://[[http|https]://]*[host[:port]]*/[prefix]*<[?[option=value]&]*
+jdbc:es://[[http|https]://]?[host[:port]]?/[prefix]?[?[option=value]&]*
```
Comment (matriv) [inline docs/reference/sql/endpoints/jdbc.asciidoc:54]: `?` means zero ore one in regex, which is what we want, but how can we differentiate from the `[?[option...]` which is the actual char `?` ? In the error message I used `\?`, but not sure. Any suggestions?

**Review by costin — APPROVED** [review]
LGTM

File: docs/reference/sql/endpoints/jdbc.asciidoc (hunk 1 above)
Comment (astefan) [reply docs/reference/sql/endpoints/jdbc.asciidoc:54]: For 0 or 1 (or `?`) there is also the `{0,1}` _more explicit_ notation, but I am not sure if we want to complicate things...

**Review by astefan — APPROVED** [review]
LGTM

**Pull request merged**